### DISCUSSION TITLE
Capacitated Minimum Spanning Tree

### DISCUSSION THREAD

**Discussion by ctc2000** [networkx/networkx#7491] [Algorithms]
I'm looking for a solution to cable routing problems common in power engineering and communications.

The input is an undirected graph, and the function outputs a minimum spanning tree with the following conditions: all branches connect to a root vertex, each branch contains (at most) x number of vertices, no edges overlap/cross. 

From what I have read, a variety of heuristics and meta-heuristics exist for this problem, but I am having trouble implementing a solution. Interested in discussing and working on this.

**Comment by mjschwenne** [comment]
Hello!

This does appear to be an interesting problem. Personally, I had not heard of a capacitated minimum spanning tree before (although it does remind me of the [cable trench problem](https://coral.ise.lehigh.edu/magh/present/informs01.pdf)). The problem description peaked my interest, so I took a look at the [Wikipedia](https://en.wikipedia.org/wiki/Capacitated_minimum_spanning_tree#cite_note-ew_alg-2) page for the problem, which cites a 1966 paper by L. R. Esau and K. C. Williams. That paper is actually pretty short (only 6 pages long), although it doesn't present any real psuedocode for the algorithm or an analysis on the bound of the approximation. I found it really cool how the original paper talks about not putting too much traffic down one of the network line and how we can use that to limit the maximum number of connections at each node. 

I guess that this is a very long way of saying that I would be interested in working on this problem with you. However, I'm currently traveling and will be for the next two weeks so I don't expect to have any time to work on it before then.

**Reply by ctc2000** [reply]
Hi! I agree that the Esau-WIlliams heuristic will be good to try. I'll give it a shot in the meantime, and look forward to learning about this problem together!
